### ISSUE TITLE
Lost ability to commission, issue with uncertified devices and attestation?

### ISSUE THREAD

**Issue by jonsmirl** [google-home/sample-app-for-matter-android#29]
Is there any solution to sharing the chip libraries with Google Home and this app? Related to this, I'm in a state right now where my Android app and devices can't talk to each other anymore because somehow the libraries are mismatched. 

```
What is this commit? It is not in the CHIP tree.
Sync matterhorn with Matter SDK at commit 0b9cae0209263edef812f4df45a4d2035c092688.

jonsmirl@ares:~/aosp/esp-matter/connectedhomeip/connectedhomeip$ git show 0b9cae0209263edef812f4df45a4d2035c092688.
fatal: ambiguous argument '0b9cae0209263edef812f4df45a4d2035c092688.': unknown revision or path not in the working tree.

My devices are using CHIP:
commit 561d23d0db215a99705ff0696e73853c8edf11b2 (HEAD, tag: v1.0.0, origin/v1.0)

```

**Comment by jonsmirl** [comment]
I'm using git main...
Did the commissioning actually complete with this Attestation failure? Apple does a popup and asks for permission to add the uncertified device.
```
2022-10-22 21:41:52.731 31292-31540 CR                      com.google.homesampleapp.default     E  mbedTLS error: ERROR - Generic error
2022-10-22 21:41:52.777 31292-31540 CTL                     com.google.homesampleapp.default     E  Failed in verifying 'Attestation Information' command received from the device: err 601. Look at AttestationVerificationResult enum to understand the errors
2022-10-22 21:41:52.777 31292-31540 CTL                     com.google.homesampleapp.default     D  Error on commissioning step 'AttestationVerification': '../../src/controller/CHIPDeviceController.cpp:962: CHIP Error 0x000000AC: Internal error'
2022-10-22 21:41:52.777 31292-31540 CTL                     com.google.homesampleapp.default     E  Failed to perform commissioning step 8
2022-10-22 21:41:52.777 31292-31540 CTL                     com.google.homesampleapp.default     D  Going from commissioning step 'AttestationVerification' with lastErr = '../../src/controller/CHIPDeviceController.cpp:962: CHIP Error 0x000000AC: Internal error' -> 'Cleanup'
2022-10-22 21:41:52.777 31292-31540 CTL                     com.google.homesampleapp.default     D  Performing next commissioning step 'Cleanup' with completion status = '../../src/controller/CHIPDeviceController.cpp:962: CHIP Error 0x000000AC: Internal error'
2022-10-22 21:41:52.777 31292-31540 CTL                     com.google.homesampleapp.default     D  Expiring failsafe on proxy 0xb40000795471a7b0
2022-10-22 21:41:52.777 31292-31540 DMG                     com.google.homesampleapp.default     D  ICR moving to [AddingComm]
2022-10-22 21:41:52.777 31292-31540 DMG                     com.google.homesampleapp.default     D  ICR moving to [AddedComma]
2022-10-22 21:41:52.777 31292-31540 IN                      com.google.homesampleapp.default     D  Prepared secure message 0xb400007b4d4d5ab8 to 0xFFFFFFFB00000000 (0)  of type 0x8 and protocolId (0, 1) on exchange 41901i with MessageCounter:7240058.
2022-10-22 21:41:52.777 31292-31540 IN                      com.google.homesampleapp.default     D  Sending encrypted msg 0xb400007b4d4d5ab8 with MessageCounter:7240058 to 0xFFFFFFFB00000000 (0) at monotonic time: 00000000201DB8C5 msec
2022-10-22 21:41:52.777 31292-31540 DMG                     com.google.homesampleapp.default     D  ICR moving to [CommandSen]
2022-10-22 21:41:52.777 31292-31540 DMG                     com.google.homesampleapp.default     D  ICR moving to [AwaitingDe]
2022-10-22 21:41:52.777 31292-31540 EM                      com.google.homesampleapp.default     D  Sending Standalone Ack for MessageCounter:104903611 on exchange 41900i
2022-10-22 21:41:52.777 31292-31540 IN                      com.google.homesampleapp.default     D  Prepared secure message 0x7954712708 to 0xFFFFFFFB00000000 (0)  of type 0x10 and protocolId (0, 0) on exchange 41900i with MessageCounter:7240059.
2022-10-22 21:41:52.777 31292-31540 IN                      com.google.homesampleapp.default     D  Sending encrypted msg 0x7954712708 with MessageCounter:7240059 to 0xFFFFFFFB00000000 (0) at monotonic time: 00000000201DB8C5 msec
2022-10-22 21:41:52.778 31292-31540 EM                      com.google.homesampleapp.default     D  Flushed pending ack for MessageCounter:104903611 on exchange 41900i
2022-10-22 21:41:52.956 31292-31540 EM                      com.google.homesampleapp.default     D  Received message of type 0x9 with protocolId (0, 1) and MessageCounter:104903612 on exchange 41901i
2022-10-22 21:41:52.956 31292-31540 EM                      com.google.homesampleapp.default     D  Found matching exchange: 41901i, Delegate: 0xb400007a2d454010
2022-10-22 21:41:52.956 31292-31540 EM                      com.google.homesampleapp.default     D  Rxd Ack; Removing MessageCounter:7240058 from Retrans Table on exchange 41901i
2022-10-22 21:41:52.956 31292-31540 EM                      com.google.homesampleapp.default     D  Removed CHIP MessageCounter:7240058 from RetransTable on exchange 41901i
2022-10-22 21:41:52.956 31292-31540 DMG                     com.google.homesampleapp.default     D  ICR moving to [ResponseRe]
2022-10-22 21:41:52.956 31292-31540 DMG                     com.google.homesampleapp.default     D  Received Command Response Data, Endpoint=0 Cluster=0x0000_0030 Command=0x0000_0001
2022-10-22 21:41:52.956 31292-31540 CTL                     com.google.homesampleapp.default     D  Failsafe disarmed
2022-10-22 21:41:52.956 31292-31540 CTL                     com.google.homesampleapp.default     D  Successfully finished commissioning step 'Cleanup'
2022-10-22 21:41:52.958 31292-31540 BaseCompletionListener  com.google.homesampleapp.default     D  onCommissioningComplete(): nodeId [8] errorCode [172]
2022-10-22 21:41:52.959 31292-31540 DMG                     com.google.homesampleapp.default     D  ICR moving to [AwaitingDe]
2022-10-22 21:41:52.959 31292-31540 EM                      com.google.homesampleapp.default     D  Sending Standalone Ack for MessageCounter:104903612 on exchange 41901i
2022-10-22 21:41:52.959 31292-31540 IN                      com.google.homesampleapp.default     D  Prepared secure message 0x7954712708 to 0xFFFFFFFB00000000 (0)  of type 0x10 and protocolId (0, 0) on exchange 41901i with MessageCounter:7240060.
2022-10-22 21:41:52.959 31292-31540 IN                      com.google.homesampleapp.default     D  Sending encrypted msg 0x7954712708 with MessageCounter:7240060 to 0xFFFFFFFB00000000 (0) at monotonic time: 00000000201DB97A msec
2022-10-22 21:41:52.959 31292-31292 AppCommiss...gRequested com.google.homesampleapp.default     D  Commissioning Step 3: Adding device to repository
2022-10-22 21:41:52.960 31292-31540 EM                      com.google.homesampleapp.default     D  Flushed pending ack for MessageCounter:104903612 on exchange 41901i
2022-10-22 21:41:52.963 31292-31292 DevicesRepository       com.google.homesampleapp.default     D  addDevice: device [# com.google.homesampleapp.Device@1835a60b
                                                                                                    date_commissioned {
                                                                                                      nanos: 960086000
                                                                                                      seconds: 1666489312
                                                                                                    }
                                                                                                    device_id: 8
                                                                                                    device_type: TYPE_UNKNOWN
                                                                                                    device_type_value: 1
                                                                                                    name: "REAL-8"
                                                                                                    product_id: "32768"
                                                                                                    vendor_id: "65521"]
2022-10-22 21:41:52.977 31292-31292 AppCommiss...gRequested com.google.homesampleapp.default     D  Commissioning Step 4: Adding device state to repository: isOnline:true isOn:false
2022-10-22 21:41:52.977 31292-31292 DevicesStateRepository  com.google.homesampleapp.default     D  addDeviceState: deviceId [8] [true] [false]
2022-10-22 21:41:52.984 31292-31292 AppCommiss...gRequested com.google.homesampleapp.default     D  Commissioning Step 5: Calling commissioningServiceDelegate.sendCommissioningComplete()
2022-10-22 21:41:52.987 31292-31292 AppCommiss...gRequested com.google.homesampleapp.default     D  OnSuccess for commissioningServiceDelegate.sendCommissioningComplete()
2022-10-22 21:44:35.408 31292-31292 Compatibil...geReporter com.google.homesampleapp.default     D  Compat change id reported: 78294732; UID 10160; state: ENABLED
2022-10-22 21:44:35.413 31292-31292 HomeViewMo...iModelFlow com.google.homesampleapp.default     D  *** devicesUiModelFlow changed ***
2022-10-22 21:44:35.414 31292-31292 HomeViewModel           com.google.homesampleapp.default     D  processDevices() deviceId: [8]}
2022-10-22 21:44:35.414 31292-31292 HomeViewModel           com.google.homesampleapp.default     D      deviceId setting its own value for state
2022-10-22 21:44:35.418 31292-31292 HomeFragment            com.google.homesampleapp.default     D  updateUi [DevicesUiModel(devices=[DeviceUiModel(device=# com.google.homesampleapp.Device@1835a60b
                                                                                                    date_commissioned {
                                                                                                      nanos: 960086000
                                                                                                      seconds: 1666489312
                                                                                                    }
                                                                                                    device_id: 8
                                                                                                    device_type: TYPE_UNKNOWN
                                                                                                    device_type_value: 1
                                                                                                    name: "REAL-8"
                                                                                                    product_id: "32768"
                                                                                                    vendor_id: "65521", isOnline=true, isOn=false)], showCodelabInfo=false, showOfflineDevices=true)]
2022-10-22 21:44:35.420 31292-31292 HomeFragment            com.google.homesampleapp.default     D  GOT result for commissioningLauncher: resultCode [-1]
2022-10-22 21:44:35.421 31292-31292 HomeViewModel           com.google.homesampleapp.default     I  Device commissioned successfully! deviceName [Jjjjhhhhjjhjhhh]
2022-10-22 21:44:35.422 31292-31292 HomeViewModel           com.google.homesampleapp.default     I  Device commissioned successfully! room [null]
```
Later on it is looking for my node 8, but it is not going to find it because it isn't in MDNS
```
2022-10-22 21:44:35.432 31292-31292 CSM                     com.google.homesampleapp.default     D  FindOrEstablishSession: PeerId = 7F67C8BD2D9F83B2:0000000000000008
2022-10-22 21:44:35.432 31292-31292 CSM                     com.google.homesampleapp.default     D  FindOrEstablishSession: No existing OperationalDeviceProxy instance found
2022-10-22 21:44:35.432 31292-31292 DIS                     com.google.homesampleapp.default     D  Resolving 7F67C8BD2D9F83B2:0000000000000008 ...
2022-10-22 21:44:35.434 31292-31292 NsdManager...ceResolver com.google.homesampleapp.default     D  Starting service resolution for '7F67C8BD2D9F83B2-0000000000000008'
```

**Comment by jonsmirl** [comment]
This might be relevant.
https://github.com/project-chip/connectedhomeip/issues/17981

**Comment by jonsmirl** [comment]
From the device side. Sample app opens the commissioning window, requests attestation, never finishes commissioning.

```
I (236894) chip[ZCL]: Received command to open commissioning window
I (236904) chip[DIS]: Updating services using commissioning mode 2
I (236904) chip[DIS]: CHIP minimal mDNS started advertising.
I (236914) chip[DIS]: Advertise operational node 5DB4B5A97C64555E-000000002DBC6E1C
I (236924) chip[DIS]: CHIP minimal mDNS configured as 'Operational device'; instance name: 5DB4B5A97C64555E-000000002DBC6E1C.
I (236934) chip[DIS]: mDNS service published: _matter._tcp
I (236944) chip[DIS]: Advertise commission parameter vendorID=65521 productID=32768 discriminator=3563/13 cm=2
I (236954) chip[DIS]: CHIP minimal mDNS configured as 'Commissionable node device'; instance name: 97E4D868D130D011.
I (236984) chip[DIS]: mDNS service published: _matterc._udp
I (236984) chip[ZCL]: Commissioning window is now open
I (236984) chip[EM]: <<< [E:51636r M:101990228 (Ack:263010965)] (S) Msg TX to 1:00000000167968C6 [555E] --- Type 0001:09 (IM:InvokeCommandResponse)
I (236994) chip[IN]: (S) Sending msg 101990228 on secure session with LSID: 17618
E (237004) chip[DL]: Long dispatch time: 129 ms, for event type 3
I (237044) app_main: Free 65784 bright 3648
I (237174) chip[EM]: >>> [E:51636r M:263010966 (Ack:101990228)] (S) Msg RX from 1:00000000167968C6 [555E] --- Type 0000:10 (SecureChannel:StandaloneAck)
I (237184) chip[EM]: >>> [E:31190r M:1086305087] (U) Msg RX from 0:0A48F5778DFACD91 [0000] --- Type 0000:20 (SecureChannel:PBKDFParamRequest)
I (237194) chip[EM]: <<< [E:31190r M:228635528 (Ack:1086305087)] (U) Msg TX to 0:0000000000000000 [0000] --- Type 0000:10 (SecureChannel:StandaloneAck)
I (237204) chip[IN]: (U) Sending msg 228635528 to IP address 'UDP:192.168.1.69%st1:46095'
I (237224) chip[EM]: <<< [E:31190r M:228635529 (Ack:1086305087)] (U) Msg TX to 0:0000000000000000 [0000] --- Type 0000:21 (SecureChannel:PBKDFParamResponse)
I (237234) chip[IN]: (U) Sending msg 228635529 to IP address 'UDP:192.168.1.69%st1:46095'
I (237544) app_main: Free 68148 bright 3648
I (237584) chip[IN]: (U) Sending msg 228635529 to IP address 'UDP:192.168.1.69%st1:46095'
I (237594) chip[EM]: >>> [E:31190r M:1086305088 (Ack:228635529)] (U) Msg RX from 0:0A48F5778DFACD91 [0000] --- Type 0000:22 (SecureChannel:PASE_Pake1)
I (237604) chip[EM]: <<< [E:31190r M:228635530 (Ack:1086305088)] (U) Msg TX to 0:0000000000000000 [0000] --- Type 0000:10 (SecureChannel:StandaloneAck)
I (237614) chip[IN]: (U) Sending msg 228635530 to IP address 'UDP:192.168.1.69%st1:46095'
I (238464) chip[EM]: <<< [E:31190r M:228635531 (Ack:1086305088)] (U) Msg TX to 0:0000000000000000 [0000] --- Type 0000:23 (SecureChannel:PASE_Pake2)
I (238474) chip[IN]: (U) Sending msg 228635531 to IP address 'UDP:192.168.1.69%st1:46095'
E (238484) chip[DL]: Long dispatch time: 891 ms, for event type 3
I (238484) chip[EM]: >>> [E:31190r M:1086305089 (Ack:228635529)] (U) Msg RX from 0:0A48F5778DFACD91 [0000] --- Type 0000:10 (SecureChannel:StandaloneAck)
I (238504) app_main: Free 66064 bright 3648
I (238634) chip[EM]: >>> [E:31190r M:1086305090 (Ack:228635531)] (U) Msg RX from 0:0A48F5778DFACD91 [0000] --- Type 0000:24 (SecureChannel:PASE_Pake3)
I (238634) chip[EM]: <<< [E:31190r M:228635532 (Ack:1086305090)] (U) Msg TX to 0:0000000000000000 [0000] --- Type 0000:10 (SecureChannel:StandaloneAck)
I (238644) chip[IN]: (U) Sending msg 228635532 to IP address 'UDP:192.168.1.69%st1:46095'
I (238664) chip[EM]: <<< [E:31190r M:228635533 (Ack:1086305090)] (U) Msg TX to 0:0000000000000000 [0000] --- Type 0000:40 (SecureChannel:StatusReport)
I (238674) chip[IN]: (U) Sending msg 228635533 to IP address 'UDP:192.168.1.69%st1:46095'
I (238684) chip[SC]: SecureSession[0x3fc97d10]: Moving from state 'kEstablishing' --> 'kActive'
I (238694) chip[SVR]: Commissioning completed session establishment step
I (238694) chip[DIS]: Updating services using commissioning mode 0
I (238704) chip[DIS]: CHIP minimal mDNS started advertising.
I (238734) chip[DIS]: Advertise operational node 5DB4B5A97C64555E-000000002DBC6E1C
I (238734) chip[DIS]: CHIP minimal mDNS configured as 'Operational device'; instance name: 5DB4B5A97C64555E-000000002DBC6E1C.
I (238744) chip[DIS]: mDNS service published: _matter._tcp
I (238744) chip[SVR]: Device completed Rendezvous process
E (238754) chip[DL]: Long dispatch time: 122 ms, for event type 3
I (238814) chip[EM]: >>> [E:31190r M:1086305091 (Ack:228635533)] (U) Msg RX from 0:0A48F5778DFACD91 [0000] --- Type 0000:10 (SecureChannel:StandaloneAck)
I (238824) chip[EM]: >>> [E:31191r M:7679599] (S) Msg RX from 0:FFFFFFFB00000000 [0000] --- Type 0001:02 (IM:ReadRequest)
I (238834) esp_matter_attribute: ********** Endpoint 0x0000's Cluster 0x0030's Attribute 0x0000 is 0 **********
I (238844) chip[EM]: <<< [E:31191r M:87726130 (Ack:7679599)] (S) Msg TX to 0:FFFFFFFB00000000 [0000] --- Type 0001:05 (IM:ReportData)
I (238854) chip[IN]: (S) Sending msg 87726130 on secure session with LSID: 17620
I (239004) app_main: Free 69144 bright 3648
I (239024) chip[EM]: >>> [E:31192r M:7679600] (S) Msg RX from 0:FFFFFFFB00000000 [0000] --- Type 0001:08 (IM:InvokeCommandRequest)
I (239024) esp_matter_command: Received command 0x0000 for enpoint 0x0000's cluster 0x00000030
I (239034) chip[FS]: GeneralCommissioning: Received ArmFailSafe (60s)
I (239034) esp_matter_attribute: ********** Endpoint 0x0000's Cluster 0x0030's Attribute 0x0000 is 3 **********
I (239044) app_main: app_attribute_update_cb 0x0 0
I (239054) app_driver: app_driver_attribute_update
I (239064) app_main: app_attribute_update_cb 0x0 0
I (239064) chip[EM]: <<< [E:31192r M:87726131 (Ack:7679600)] (S) Msg TX to 0:FFFFFFFB00000000 [0000] --- Type 0001:09 (IM:InvokeCommandResponse)
I (239074) chip[IN]: (S) Sending msg 87726131 on secure session with LSID: 17620
I (239094) chip[EM]: >>> [E:31191r M:7679601 (Ack:87726130)] (S) Msg RX from 0:FFFFFFFB00000000 [0000] --- Type 0000:10 (SecureChannel:StandaloneAck)
I (239224) chip[EM]: >>> [E:31193r M:7679602] (S) Msg RX from 0:FFFFFFFB00000000 [0000] --- Type 0001:08 (IM:InvokeCommandRequest)
I (239234) esp_matter_command: Received command 0x0002 for enpoint 0x0000's cluster 0x00000030
I (239234) chip[DL]: NVS set: chip-config/reg-location = 0 (0x0)
I (239254) chip[DL]: NVS set: chip-config/country-code = "XX"
I (239254) esp_matter_attribute: ********** Endpoint 0x0000's Cluster 0x0030's Attribute 0x0000 is 4 **********
I (239264) app_main: app_attribute_update_cb 0x0 0
I (239264) app_driver: app_driver_attribute_update
I (239274) app_main: app_attribute_update_cb 0x0 0
I (239274) chip[EM]: <<< [E:31193r M:87726132 (Ack:7679602)] (S) Msg TX to 0:FFFFFFFB00000000 [0000] --- Type 0001:09 (IM:InvokeCommandResponse)
I (239294) chip[IN]: (S) Sending msg 87726132 on secure session with LSID: 17620
I (239304) chip[EM]: >>> [E:31192r M:7679603 (Ack:87726131)] (S) Msg RX from 0:FFFFFFFB00000000 [0000] --- Type 0000:10 (SecureChannel:StandaloneAck)
I (239324) ROUTE_HOOK: Received RIO
I (239324) ROUTE_HOOK: prefix FDD5:204F:C807:: lifetime 1800

I (239434) chip[EM]: >>> [E:31194r M:7679604] (S) Msg RX from 0:FFFFFFFB00000000 [0000] --- Type 0001:08 (IM:InvokeCommandRequest)
I (239434) esp_matter_command: Received command 0x0002 for enpoint 0x0000's cluster 0x0000003E
I (239444) chip[ZCL]: OpCreds: Certificate Chain request received for PAI
I (239454) chip[EM]: <<< [E:31194r M:87726133 (Ack:7679604)] (S) Msg TX to 0:FFFFFFFB00000000 [0000] --- Type 0001:09 (IM:InvokeCommandResponse)
I (239464) chip[IN]: (S) Sending msg 87726133 on secure session with LSID: 17620
I (239474) chip[EM]: >>> [E:31193r M:7679605 (Ack:87726132)] (S) Msg RX from 0:FFFFFFFB00000000 [0000] --- Type 0000:10 (SecureChannel:StandaloneAck)
I (239504) app_main: Free 69144 bright 3648
I (239534) chip[EM]: >>> [E:31195r M:7679606] (S) Msg RX from 0:FFFFFFFB00000000 [0000] --- Type 0001:08 (IM:InvokeCommandRequest)
I (239534) esp_matter_command: Received command 0x0002 for enpoint 0x0000's cluster 0x0000003E
I (239544) chip[ZCL]: OpCreds: Certificate Chain request received for DAC
I (239554) chip[EM]: <<< [E:31195r M:87726134 (Ack:7679606)] (S) Msg TX to 0:FFFFFFFB00000000 [0000] --- Type 0001:09 (IM:InvokeCommandResponse)
I (239564) chip[IN]: (S) Sending msg 87726134 on secure session with LSID: 17620
I (239574) chip[EM]: >>> [E:31194r M:7679607 (Ack:87726133)] (S) Msg RX from 0:FFFFFFFB00000000 [0000] --- Type 0000:10 (SecureChannel:StandaloneAck)
I (239744) chip[EM]: >>> [E:31196r M:7679608] (S) Msg RX from 0:FFFFFFFB00000000 [0000] --- Type 0001:08 (IM:InvokeCommandRequest)
I (239744) esp_matter_command: Received command 0x0000 for enpoint 0x0000's cluster 0x0000003E
I (239754) chip[ZCL]: OpCreds: Received an AttestationRequest command
I (239764) chip[EM]: <<< [E:31196r M:87726135 (Ack:7679608)] (S) Msg TX to 0:FFFFFFFB00000000 [0000] --- Type 0000:10 (SecureChannel:StandaloneAck)
I (239774) chip[IN]: (S) Sending msg 87726135 on secure session with LSID: 17620
I (239954) chip[ZCL]: OpCreds: AttestationRequest successful.
I (239954) chip[EM]: <<< [E:31196r M:87726136 (Ack:7679608)] (S) Msg TX to 0:FFFFFFFB00000000 [0000] --- Type 0001:09 (IM:InvokeCommandResponse)
I (239964) chip[IN]: (S) Sending msg 87726136 on secure session with LSID: 17620
E (239974) chip[DL]: Long dispatch time: 231 ms, for event type 3
I (239974) chip[EM]: >>> [E:31195r M:7679609 (Ack:87726134)] (S) Msg RX from 0:FFFFFFFB00000000 [0000] --- Type 0000:10 (SecureChannel:StandaloneAck)
I (240004) app_main: Free 69144 bright 3648
I (240144) chip[EM]: >>> [E:31197r M:7679610] (S) Msg RX from 0:FFFFFFFB00000000 [0000] --- Type 0001:08 (IM:InvokeCommandRequest)
I (240154) esp_matter_command: Received command 0x0000 for enpoint 0x0000's cluster 0x00000030
I (240164) chip[FS]: GeneralCommissioning: Received ArmFailSafe (0s)
I (240164) chip[FS]: Fail-safe timer expired
I (240174) chip[EM]: <<< [E:31197r M:87726137 (Ack:7679610)] (S) Msg TX to 0:FFFFFFFB00000000 [0000] --- Type 0001:09 (IM:InvokeCommandResponse)
I (240184) chip[IN]: (S) Sending msg 87726137 on secure session with LSID: 17620
I (240194) chip[EM]: >>> [E:31196r M:7679611 (Ack:87726136)] (S) Msg RX from 0:FFFFFFFB00000000 [0000] --- Type 0000:10 (SecureChannel:StandaloneAck)
E (240204) chip[SVR]: Failsafe timer expired
I (240214) chip[SC]: SecureSession[0x3fc97d10]: Moving from state 'kActive' --> 'kPendingEviction'
E (240224) chip[SVR]: Commissioning failed (attempt 1): 32
I (240234) chip[DIS]: Updating services using commissioning mode 2
I (240234) chip[DIS]: CHIP minimal mDNS started advertising.
I (240244) chip[DIS]: Advertise operational node 5DB4B5A97C64555E-000000002DBC6E1C
I (240254) chip[DIS]: CHIP minimal mDNS configured as 'Operational device'; instance name: 5DB4B5A97C64555E-000000002DBC6E1C.
I (240264) chip[DIS]: mDNS service published: _matter._tcp
I (240264) chip[DIS]: Advertise commission parameter vendorID=65521 productID=32768 discriminator=3563/13 cm=2
I (240284) chip[DIS]: CHIP minimal mDNS configured as 'Commissionable node device'; instance name: 97E4D868D130D011.
I (240304) chip[DIS]: mDNS service published: _matterc._udp
E (240314) chip[ZCL]: OpCreds: Got FailSafeTimerExpired
E (240314) chip[ZCL]: OpCreds: Proceeding to FailSafeCleanup on fail-safe expiry!
I (240314) chip[TS]: Pending Last Known Good Time: 2022-10-23T14:51:23
I (240324) chip[TS]: Previous Last Known Good Time: 2022-10-23T14:51:23
I (240334) chip[TS]: Reverted Last Known Good Time to previous value
I (240344) esp_matter_attribute: ********** Endpoint 0x0000's Cluster 0x0030's Attribute 0x0000 is 0 **********
I (240354) app_main: app_attribute_update_cb 0x0 0
I (240354) app_driver: app_driver_attribute_update
I (240364) app_main: app_attribute_update_cb 0x0 0
E (240364) chip[DL]: Long dispatch time: 159 ms, for event type 32782
I (240374) chip[FS]: Fail-safe cleanly disarmed
E (240384) chip[IN]: Data received on an unknown session (LSID=17620). Dropping it!
I (240504) app_main: Free 69888 bright 3648
I (241004) app_main: Free 69884 bright 3648
I (241504) app_main: Free 69884 bright 3648
```

**Comment by jonsmirl** [comment]
I don't have access to the CHIP doc, so I am reverting to reading the source code, the CHIP JNI library implements setUseJavaCallbackForNOCRequest() and the Google implementation of ChipDeviceController does not have that method.  My guess is that method is used to set the callback asking if it is ok to allow uncertified devices.

**Comment by pierredelisle** [comment]
As mentioned in https://github.com/google-home/sample-app-for-matter-android/issues/28,

> Due to the difference in the Matter SDK SHA that Google's Developer Preview rolls up to, vs where the SDK is today, we have multiple minor version releases available (1.0, 1.1, 1.2). Each is intended for a different purpose, and we should clarify this in the README until our public developer support fully aligns with the Matter 1.0 spec.

Because of that, it is critical that both the client (GHSAFM) and the device use a compatible version of the Matter SDK.
If you look at this Matter SDK commit:

https://github.com/project-chip/connectedhomeip/commit/faad9e179293775110f09692cd1c28e0d71b792f

you will see that the CD Signing Certificates have been updated to the CSA Official ones. If the client and the device have mismatched SDKs when it comes to these signing certs, then you will see the error 601.

**Comment by jonsmirl** [comment]
So what Matter commit do I need to be on? I just want to work on my Android app and not have Google Home and Matter break on me.

**Comment by jonsmirl** [comment]
I moved my devices over to Matter master and I can talk to them again. So I guess this v1.0 tag in Matter is meaningless?

**Comment by pierredelisle** [comment]
For things to work, both the app and the device must rely on versions of the “Matter SDK” and the “Home Mobile SDK for Android” that are compatible.

For Google’s current “Developer Preview”, please use GHSAFM 1.0.3 that is in sync with the “Home Mobile SDK for Android” that’s currently available via Google Play Services on Android phones. The Matter SDK that is used with v1.0.3 is branch interop_testing_te9.

At regular intervals and for internal testing, we’re trying to sync the app with the latest changes on the Matter SDK until we are ready to release our official support for Matter 1.0. That’s why we released v1.1.0 and v1.2.0. Note that each one of these releases clearly states the Matter SDK commit against which they were sync’ed. In retrospect, not sure it was a good idea to create these releases as they may lead to confusion. I’ll refrain to do so until we are ready to release our official support for Matter 1.0. 

Jeff has now updated the Releases section of the [README](https://github.com/google-home/sample-app-for-matter-android#releases) to clarify that v1.0.3 is the version of the app to use with the Developer Preview.

**Comment by jakubdybczak** [comment]
@pierredelisle is there estimated date for support of Matter 1.0? In both of GHSAFM and SDK included within Google Play Services?
